### ISSUE TITLE
VS UI deadlock initializing credential providers during restore

### ISSUE THREAD

**Issue by pspill** [NuGet/Home#2361]
This hits because the NuGet Credential Service a) initializes all providers lazily as needed, b) does the MEF import on provider initialization, and c) switches to the UI thread in order to do b.

If you haven’t done any other NuGet operation that causes the providers to be loaded, and kick of a build, a UI dialog pops up to show NuGet restore progress, which is contending with loading the providers.

If you hit cancel in the restore packages dialog this unblocks provider initialization, and you can now build and restore from all package sources successfully, but we still can’t ship like this.

The fix is to not do lazy init and just build all the providers on NuGet startup.

**Comment by rrelyea** [comment]
Fixed merged into 3.4 for vs2015.2 on Friday 3/18.
